extmod/modbluetooth.c

```diff
@@ -951,6 +1016,23 @@ STATIC mp_obj_t bluetooth_ble_invoke_irq(mp_obj_t none_in) {
             // conn_handle, value_handle, status
             ringbuf_extract(&o->ringbuf, data_tuple, 3, 0, NULL, 0, NULL, NULL);
         #endif // MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE
+        #if MICROPY_PY_BLUETOOTH_ENABLE_L2CAP_COC
+        } else if (event == MP_BLUETOOTH_IRQ_L2CAP_COC_ACCEPT) {
```

Comment:
why is this code needed, I thought L2CAP wasn't supported if SYNC_EVENTS is disabled?

Reply:
Initially I planned to support it in both modes, but the ACCEPT event now needs to be synchronous, so I've removed all L2CAP support for ringbuffer-mode.